GameEngine/GameState.cs

```diff
@@ -614,23 +627,40 @@ public List<Character> GetCharactersInLocation(Guid locationTrackingId, bool inc
         {
             var characters = CharacterLocations
                 .Where(kvp => kvp.Value == locationTrackingId) // Where location is the one passed in
-                .Select(kvp => GetCharacter(kvp.Key));
+                .Select(kvp => GetCharacter(kvp.Key)).ToList();
 
             // remove player if needed.
             if (!includePlayer)
             {
-                characters = characters
-                    .Where(c => c.TrackingId != PlayerTrackingId);
+                foreach(var playerId in PlayerTrackingIds)
+                {
+                    var player = GetCharacter(playerId);
+                    characters.Remove(player);
+                }
             }
 
             return characters.ToList();
         }
 
-        public Location GetPlayerCharacterLocation()
+        public List<Location> GetPlayerCharacterLocations()
         {
-            return GetCharacterLocation(PlayerTrackingId);
+            return GetCharacterLocation(PlayerTrackingIds);
```

Comment:
This doesn't seem like an accurate enough response. It doesn't tell what player is at what location.

Reply:
that game functions, and it knows where each player is.